src/sources/datadog_agent/mod.rs

```diff
@@ -188,53 +189,55 @@ impl SourceConfig for DatadogAgentConfig {
     }
 
     fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<Output> {
-        let log_namespace = global_log_namespace.merge(self.log_namespace);
-
-        let definition = match log_namespace {
-            LogNamespace::Legacy => {
-                match self.decoding {
-                    // See: `LogMsg` struct.
-                    DeserializerConfig::Bytes => self
-                        .decoding
-                        .schema_definition(log_namespace)
-                        .with_field("message", Kind::bytes(), Some("message"))
-                        .with_field("status", Kind::bytes(), Some("severity"))
-                        .with_field("timestamp", Kind::timestamp(), Some("timestamp"))
-                        .with_field("hostname", Kind::bytes(), Some("host"))
-                        .with_field("service", Kind::bytes(), Some("service"))
-                        .with_field("ddsource", Kind::bytes(), Some("source"))
-                        .with_field("ddtags", Kind::bytes(), Some("tags")),
-
-                    // JSON deserializer can overwrite existing fields at runtime, so we have to treat
-                    // those events as if there is no known type details we can provide, other than the
-                    // details provided by the generic JSON schema definition.
-                    DeserializerConfig::Json => self.decoding.schema_definition(log_namespace),
-
-                    // Syslog deserializer allows for arbritrary "structured data" that can overwrite
-                    // existing fields, similar to the JSON deserializer.
-                    //
-                    // See also: https://datatracker.ietf.org/doc/html/rfc5424#section-6.3
-                    #[cfg(feature = "sources-syslog")]
-                    DeserializerConfig::Syslog => self.decoding.schema_definition(log_namespace),
-
-                    DeserializerConfig::Native => self.decoding.schema_definition(log_namespace),
-                    DeserializerConfig::NativeJson => {
-                        self.decoding.schema_definition(log_namespace)
-                    }
-                    DeserializerConfig::Gelf => self.decoding.schema_definition(log_namespace),
-                }
-            }
-            LogNamespace::Vector => self
-                .decoding
-                .schema_definition(log_namespace)
-                .with_metadata_field("message", Kind::bytes())
-                .with_metadata_field("status", Kind::bytes())
-                .with_metadata_field("timestamp", Kind::timestamp())
-                .with_metadata_field("hostname", Kind::bytes())
-                .with_metadata_field("service", Kind::bytes())
-                .with_metadata_field("ddsource", Kind::bytes())
-                .with_metadata_field("ddtags", Kind::bytes()),
-        };
+        let source_name = "datadog_agent";
```

Comment:
Optional improvement: Noticed this raw string is used in this file in a few other places. Could add a const at the top of the file near the existing consts (`LOGS` `METRICS` `TRACES`) for "datadog_agent" , and update the references in the file to use it instead of defining one here.

```Rust
const NAME: &str = "datadog_agent";
```

Reply:
Hmm, I guess I should just use the already existing function on the `SourceConfig` trait called `source_type` which already returns the same thing :sweat_smile:

Reply:
@spencergilbert and this makes the last `.with_source_metadata` line long enough to allow newlines :partying_face:

Reply:
fixed in 4901b06d1f8fe13d799df12395ef0c228ba34d2c